### PR TITLE
fix [buildPlugin] define specific JDK tool for container agents

### PR DESCRIPTION
**Pull request by dduportal**
This PR is an attempt at fixing #230 and #228.

It ensures that the JAVA_HOME is correctly set without the "tool".

There are 2 cases:

- When it is a container, it assumes that its either a `jnlp-maven-8` or a `jnlp-maven-11`, which are based on the Maven's Docker image (the temurin version) with a JDK in `/opt/java/openjdk` (as per. https://github.com/adoptium/containers/blob/main/8/jdk/ubuntu/Dockerfile.releases.full#L63). The other JNLP templates (`jnlp-ruby`, `jnlp-node` and `jnlp-alpine`) are not expected to be used by this library (`buildPlugin`)

-  Otherwise if it is a VM, then it uses the new `/opt/jdk-<VERSION>`

Please note that the "tools" are still needed for static agents (such as the s390x on ci.jenkins.io or agent-1 on trusted.ci)

### CODE REVIEW

File: vars/infra.groovy [hunk 1]
```diff
@@ -127,14 +127,20 @@ Object runWithMaven(String command, String jdk = 8, List<String> extraEnv = null
...
-Object runWithJava(String command, String jdk = 8, List<String> extraEnv = null, Boolean addToolEnv = true) {
+Object runWithJava(String command, String jdk = 8, List<String> extraEnv = null, Boolean addToolEnv = true, Boolean useContainerAgent = false) {
     List<String> env = [];
     if(addToolEnv) {
-        String jdkTool = "jdk${jdk}"
-        env = [
-            "JAVA_HOME=${tool jdkTool}",
-            'PATH+JAVA=${JAVA_HOME}/bin',
-        ]
+        if(useContainerAgent){
```
[the remark this answers is not in the document]
Comment (dduportal) [reply vars/infra.groovy]: implemented with the `fileExists`, WDYT?

File: vars/infra.groovy [hunk 2]
```diff
@@ -127,14 +127,22 @@ Object runWithMaven(String command, String jdk = 8, List<String> extraEnv = null
...
+                'JAVA_HOME=/opt/java/openjdk', // https://github.com/adoptium/containers/blob/main/8/jdk/ubuntu/Dockerfile.releases.full#L63
+                'PATH+JAVA=${JAVA_HOME}/bin',
+            ]
+        } else if(fileExists("/opt/jdk-${jdk}/bin/java")) {
+            env = [
+                "JAVA_HOME=/opt/jdk-${jdk}",
+                'PATH+JAVA=${JAVA_HOME}/bin',
+            ]
+        } else {
+            echo "WARNING: environment variables JAVA_HOME and PATH not updated, because no java binary found in either /opt/java/openjdk/bin/java or /opt/jdk-${jdk}/bin/java."
```
Comment (timja) [inline vars/infra.groovy:144]: windows?

File: vars/buildPlugin.groovy [hunk 3]
```diff
@@ -140,7 +140,7 @@ def call(Map params = [:]) {
                                 }
 
                                 try {
-                                    infra.runWithJava(command, jdk, null, addToolEnv)
+                                    infra.runWithJava(command, jdk, null, addToolEnv, useContainerAgent)
```
Comment (timja) [inline vars/buildPlugin.groovy:143]: revert

File: test/groovy/mock/Infra.groovy [hunk 4]
```diff
@@ -24,7 +24,7 @@ class Infra implements Serializable {
     return runWithJava(command, jdk, extraEnv, addToolEnv)
   }
 
-  public Object runWithJava(String command, String jdk = null, List<String> extraEnv = null, Boolean addToolEnv = null) {
+  public Object runWithJava(String command, String jdk = null, List<String> extraEnv = null, Boolean addToolEnv = null, Boolean useContainerAgent = null) {
```
Comment (timja) [inline test/groovy/mock/Infra.groovy:27]: revert

**Pull request merged**